src/pkcs11/core_pkcs11_test.c

```diff
@@ -0,0 +1,2647 @@
+
+/* Thread function of AFQP_FindObjectMultiThread test cases for EC and RSA.
+ * Repeatedly tries to find previously provisioned private key and certificate. */
+static void prvFindObjectMultiThreadTask( void * pvParameters )
+{
+    MultithreadTaskParams_t * pxMultiTaskParam = pvParameters;
+    uint32_t xCount;
+    CK_RV xResult;
+    CK_OBJECT_HANDLE xHandle;
+    CK_SESSION_HANDLE xSession;
+
+    memcpy( &xSession, pxMultiTaskParam->pvTaskData, sizeof( CK_SESSION_HANDLE ) );
+
+    for( xCount = 0; xCount < PKCS11_TEST_MULTI_THREAD_LOOP_COUNT; xCount++ )
+    {
+        xResult = xFindObjectWithLabelAndClass( xSession,
+                                                PKCS11_TEST_LABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+                                                sizeof( PKCS11_TEST_LABEL_DEVICE_PRIVATE_KEY_FOR_TLS ) - 1,
+                                                CKO_PRIVATE_KEY,
+                                                &xHandle );
+
+        if( xResult != CKR_OK )
+        {
+            TEST_PRINTF( "FindObject multithreaded task failed to find private key.  Error: %ld  Count: %d \r\n", xResult, xCount );
+            break;
+        }
+
+        if( ( xHandle == CK_INVALID_HANDLE ) )
+        {
+            TEST_PRINTF( "FindObject multi-thread task failed to find private key.  Invalid object handle returned.  Count: %d \r\n", xCount );
+            xResult = CKR_OBJECT_HANDLE_INVALID; /* Mark xResult so that test fails. */
+            break;
+        }
+
+        xResult = xFindObjectWithLabelAndClass( xSession,
+                                                PKCS11_TEST_LABEL_DEVICE_CERTIFICATE_FOR_TLS,
+                                                sizeof( PKCS11_TEST_LABEL_DEVICE_CERTIFICATE_FOR_TLS ) - 1,
+                                                CKO_CERTIFICATE,
+                                                &xHandle );
+
+        if( xResult != CKR_OK )
+        {
+            TEST_PRINTF( "FindObject multithreaded task failed to find certificate.  Error: %ld  Count: %d \r\n", xResult, xCount );
+            break;
+        }
+
+        if( ( xHandle == CK_INVALID_HANDLE ) )
+        {
+            TEST_PRINTF( "FindObject multi-thread task failed to find certificate.  Invalid object handle returned. Count: %d \r\n", xCount );
+            xResult = CKR_OBJECT_HANDLE_INVALID; /* Mark xResult so that test fails. */
+            break;
+        }
+    }
+
+    /* Report the result of the loop. */
+    pxMultiTaskParam->xTestResult = xResult;
+}
+
+/*-----------------------------------------------------------*/
+
+/* Different session trying to find token objects. */
+TEST( Full_PKCS11_RSA, AFQP_FindObjectMultiThread )
+{
+    CK_RV xResult;
+    uint32_t xTaskNumber;
+    CK_SESSION_HANDLE xSessionHandle[ PKCS11_TEST_MULTI_THREAD_TASK_COUNT ];
+    CK_OBJECT_HANDLE xPrivateKey;
+    CK_OBJECT_HANDLE xPublicKey;
+    CK_OBJECT_HANDLE xCertificate;
+

```

Comment:
Need to set `xGlobalTaskParams.xTestResult` to zero before running the tests.

Reply:
This will be udpated in new commit. Thank you for the comment.